apps/server/src/modules/oauth/controller/oauth-sso.controller.ts

```diff
@@ -22,22 +19,8 @@ export class OauthSSOController {
 		@Res() res: Response,
 		@Param('systemid', ParseObjectIdPipe) systemid: string
 	): Promise<unknown> {
-		let oauthResponse: OAuthResponse;
-		const HOST = Configuration.get('HOST') as string;
-		try {
-			oauthResponse = await this.oauthUc.startOauth(query, systemid);
-			if (oauthResponse.jwt) {
-				res.cookie('jwt', oauthResponse.jwt);
-				const idToken = oauthResponse.idToken as string;
-				const logoutEndpoint = oauthResponse.logoutEndpoint as string;
-				return res.redirect(`${logoutEndpoint}?id_token_hint=${idToken}&post_logout_redirect_uri=${HOST}/dashboard`);
-			}
-		} catch (error) {
-			this.logger.error(error);
-			oauthResponse = new OAuthResponse();
-			if (error instanceof OAuthSSOError) oauthResponse.errorcode = error.errorcode;
-			else oauthResponse.errorcode = 'oauth_login_failed';
-		}
-		return res.redirect(`${HOST}/login?error=${oauthResponse.errorcode as string}`);
+		const oauthResponse = await this.oauthUc.startOauth(query, systemid);
```

Comment:
Can be oauthResponse null?

Reply:
oauthResponse cannot be null. It has either a 
complete response or
an OAuthSsoError or
as the else case of the catch block a default error.
In any case oauthResponse has some entry and is therefor not null.